### PR TITLE
State transitions for presence event engine

### PR DESCRIPTION
**Pull request by Xavrax**

### CODE REVIEW

File: src/Api/PubnubApi/EventEngine/Presence/States/CooldownState.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            return ev switch 
+            {
+                Events.JoinedEvent e => new HeartbeatingState()
+                {
+                    Input = e.Input != this.Input ? this.Input + e.Input : this.Input,
+                },
+                Events.LeftEvent e => HandleLeftEvent(e),
+                Events.LeftAllEvent e => new InactiveState(),
+                Events.HeartbeatSuccessEvent e => new CooldownState()
```
Comment (mohitpubnub) [inline src/Api/PubnubApi/EventEngine/Presence/States/CooldownState.cs:26]: Do we need to handle `HeratbeatSuccess` when we are on `Cooldown` state ?

**Review by mohitpubnub — COMMENTED** [review]
overall it looks good.

File: src/Api/PubnubApi/EventEngine/Presence/States/CooldownState.cs (hunk 1 above)
Comment (Xavrax) [reply src/Api/PubnubApi/EventEngine/Presence/States/CooldownState.cs:26]: Good catch. Probably I've added that in favour of Heartbeating and Reconnecting! I will change that after the compiling fix.

**Comment by mohitpubnub** [comment]
Summary:
@Xavrax 
the [issue](https://github.com/pubnub/c-sharp/actions/runs/7652704390/job/20853025792#step:6:38) occurs as compiler is not able to recognise C# syntax used for event engine implementation.
starting from [56ebedd](https://github.com/pubnub/c-sharp/pull/203/commits/56ebedd5571a1851a317e70b3985d8d0708b6481). Attempted couple of valid values for LangVersion mentioned [here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#c-language-version-reference) in official docs.
now TODO: need to check why CI is not able to recognise the syntax. btw its not issue at local environment... Followed several SO threads but it doesn't recognise sepcific C# versions that we need.

**Review by mohitpubnub — APPROVED** [review]
LGTM!

**Pull request merged**